README.md

```diff
@@ -159,29 +159,59 @@ content type.
 becomes...
 
 ```eex
-<%= content(@conn, "Image identifier", :background_image, do: "http://placekitten.com/200/300")
-%>
+<%= content(@conn, "Image identifier", :background_image, do: "http://placekitten.com/200/300") %>
 ```
 
 ### Image Uploads
 
+Thesis offers support for a few different ways to handle image uploads: store files in the database,
+point to an uploader/adapter inside your custom app, or use one of the prebuilt adapters.
+
+##### Store Files in Database
+For smaller websites and/or website that are hosted on the cloud, thesis offers a no-setup-required image uploader.
+Files are stored in a separate table and contain all of the needed metadata (name, file type, and blobs themselves).
+
+To enable, add this in your config/config.exs file:
+
+```elixir
+config :thesis,
+  uploader: Thesis.RepoUploader
+```
+
+##### User Your Own Uploader Module
+
+If you already set up file uploads in your custom app, point thesis to a module that can handle a `%Plug.Upload{}`
+struct.
+
+```elixir
+config :thesis,
+  uploader: <MyApp>.<CustomUploaderModule>
+```
+
+The module should have an `upload/1` function that accepts a `%Plug.Upload{}` struct. This function should return
+an image url string, image path string, or a blank string. You can view
+[/lib/thesis/uploaders/repo_uploader.ex](https://github.com/infinitered/thesis-phoenix/blob/master/lib/thesis/uploaders/repo_uploader.ex)
+as an example.
```

Comment:
Let's use a behaviour to enforce this. Also, we should be returning a tuple rather than a string.

`{:ok, "path/to/file.jpg"}`